File: web/lib/playback-state.ts
import { DisplayBuffer } from "./display-buffer";

export class PlaybackState {
  currentTick: number;
  isPlaying: boolean;
  tickAccumulatorMicros: number;
  displayBuffer: DisplayBuffer;

  constructor(displayBuffer: DisplayBuffer) {
    this.displayBuffer = displayBuffer;
    this.currentTick = 0;
    this.isPlaying = false;
    this.tickAccumulatorMicros = 0;
  }

  reset(): void {
    this.currentTick = 0;
    this.tickAccumulatorMicros = 0;
    this.isPlaying = false;
  }
}

File: android/lib/src/main/kotlin/io/grin/lib/PlaybackState.kt
package io.grin.lib

class PlaybackState(val displayBuffer: DisplayBuffer) {
    var currentTick: Long = 0
    var isPlaying: Boolean = false
    var tickAccumulatorMicros: Long = 0

    fun reset() {
        currentTick = 0
        tickAccumulatorMicros = 0
        isPlaying = false
    }
}
